hat/assets/js/apps/Iaso/domains/dataSources/components/sync/ConfirmSyncPreview.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { FunctionComponent } from 'react';
+import { Button, Typography, Box } from '@mui/material';
+import { useSafeIntl } from 'bluesquare-components';
+import MESSAGES from '../../messages';
+import { SyncResponse } from '../../types/sync';
+
+type Props = {
+    jsonDiffResult?: SyncResponse;
+    isDisabled: boolean;
+    handleSeePreview: () => void;
+};
+export const ConfirmSyncPreview: FunctionComponent<Props> = ({
+    jsonDiffResult,
+    isDisabled,
+    handleSeePreview,
+}) => {
+    const { formatMessage } = useSafeIntl();
+
+    return (
+        <>
+            <Box display="flex" justifyContent="center" width="100%" my={2}>
+                <Button
+                    onClick={handleSeePreview}
+                    color="primary"
+                    variant="outlined"
+                    disabled={isDisabled}
+                >
+                    {formatMessage(MESSAGES.syncPreview)}
+                </Button>
+            </Box>
+            {jsonDiffResult && (
+                <Box
+                    sx={{
+                        bgcolor: 'grey.200',
+                        p: 2,
+                        borderRadius: 1,
+                        mb: 3,
+                        textAlign: 'center',
+                    }}
+                >
+                    <Box
+                        sx={{
+                            display: 'flex',
+                            gap: 4,
+                            justifyContent: 'center',
+                        }}
+                    >
+                        <Box>
+                            <Typography variant="subtitle2">
+                                {formatMessage(MESSAGES.count_create)}
+                            </Typography>
+                            <Typography variant="h4">
+                                {jsonDiffResult.count_create}
+                            </Typography>
+                        </Box>
+                        <Box>
+                            <Typography variant="subtitle2">
+                                {formatMessage(MESSAGES.count_update)}
+                            </Typography>
+                            <Typography variant="h4">
+                                {jsonDiffResult.count_update}
+                            </Typography>
+                        </Box>
+                    </Box>
+                </Box>
+            )}
+            {!jsonDiffResult && (
+                <Typography
+                    variant="body2"
+                    component="span"
+                    sx={{
+                        display: 'flex',
+                        justifyContent: 'center',
+                        width: '100%',
+                        color: 'warning.main',
+                    }}
+                >
+                    {formatMessage(MESSAGES.syncMessage)}
```

Comment:
Once triggered, the "Preview Change Request(s)" button should be disabled to avoid creating multiple synchronization objects in DB.

Ideally there should be 2 distinct API calls:

- the first creates the synchronization object with a name
- the second uses the ID of the previously created object to trigger the preview

![disable](https://github.com/user-attachments/assets/803828af-8efb-45f8-b4ec-5931a4fbe1f4)

Reply:
One last thing: the change requests will be created for the source version specified in the `Source version to update` field.

It would be great to have a message explaining this to the user!

Reply:
done